packages/wallets/src/api/client.ts

```diff
@@ -0,0 +1,302 @@
+
+import type {
+    ClientOptions,
+    CreateWalletDto,
+    WalletV1Alpha2ResponseDto,
+    CreateTransactionDto,
+    SubmitApprovalDto,
+    WalletsV1ControllerGetTransaction4Response,
+    WalletsV1Alpha2TransactionResponseDto,
+    CreateSignatureRequestDto,
+    WalletsV1Alpha2SignatureResponseDto,
+    WalletsV1Alpha2TransactionsResponseDto,
+    Nftevm,
+    Nftsol,
+    WalletBalanceResponseDto,
+} from "./gen/types.gen";
+import {
+    walletsV1ControllerCreateWallet4,
+    walletsV1ControllerGetWalletByLocator4,
+    walletsV1ControllerCreateTransaction4,
+    walletsV1ControllerSubmitApprovals4,
+    walletsV1ControllerGetTransaction4,
+    walletsV1ControllerCreateSignatureRequest4,
+    walletsV1ControllerSubmitSignatureApprovals4,
+    walletsV1ControllerGetSignature4,
+    walletsV1ControllerGetTransactionsWithoutChain4,
+    fetchContentFromWallet,
+    balanceControllerGetBalanceForLocator2,
+} from "./gen/sdk.gen";
+
+type CreateWalletParams = CreateWalletDto;
+type CreateWalletResponse = WalletV1Alpha2ResponseDto;
+type GetWalletResponse = WalletV1Alpha2ResponseDto;
+
+type CreateTransactionParams = CreateTransactionDto;
+type CreateTransactionResponse = WalletsV1Alpha2TransactionResponseDto;
+type ApproveTransactionParams = SubmitApprovalDto;
+type ApproveTransactionResponse = WalletsV1ControllerGetTransaction4Response;
+type GetTransactionResponse = WalletsV1Alpha2TransactionResponseDto;
+
+type CreateSignatureParams = CreateSignatureRequestDto;
+type CreateSignatureResponse = WalletsV1Alpha2SignatureResponseDto;
+type ApproveSignatureParams = SubmitApprovalDto;
+type ApproveSignatureResponse = WalletsV1Alpha2SignatureResponseDto;
+type GetSignatureResponse = WalletsV1Alpha2SignatureResponseDto;
+
+type GetTransactionsResponse = WalletsV1Alpha2TransactionsResponseDto;
+type GetNftsResponse = Nftevm | Nftsol;
+type GetBalanceResponse = WalletBalanceResponseDto;
+
+class ApiClient {
+    private readonly client: Client;
+
+    constructor(
+        private readonly apiKey: string,
+        private readonly jwt?: string
+    ) {
+        const validationResult = validateAPIKey(this.apiKey);
+        if (!validationResult.isValid) {
+            throw new Error("Invalid API key");
+        }
+        if (validationResult.usageOrigin === APIKeyUsageOrigin.CLIENT && !jwt) {
+            throw new Error("JWT is required for client API key");
+        }
+
+        const baseUrl = environmentToCrossmintBaseURL(validationResult.environment);
+        this.client = createClient(
+            createConfig<ClientOptions>({
+                baseUrl,
+            })
+        );
+    }
+
+    async createWallet(params: CreateWalletParams): Promise<CreateWalletResponse> {
+        const response = await walletsV1ControllerCreateWallet4({
+            client: this.client,
+            body: params,
+            headers: {
+                "X-API-KEY": this.apiKey,
+            },
```

Comment:
there seems to be some redundancy here, as we will need to pass this on every call

Any way to abstract that out s e.g. the client takes care of adding this header?

Reply:
this is the limitation of this tool (& multiple others) i'm afraid.

I've refactored it a bit in https://github.com/Crossmint/crossmint-sdk/pull/956/commits/a4adf98aefb1fb01ee16c7035cdf710e00faf9c6 by adding a `getHeaders` helper

Reply:
we're using `CrossmintApiClient` now, so no longer an issue